### PR TITLE
Invalid `dsc_scripts` should fail the run

### PR DESCRIPTION
**Pull request by NimishaS**
w.r.t https://github.com/chef/chef/issues/4534

### CODE REVIEW

**Comment by NimishaS** [comment]
The warning specified in https://github.com/chef/chef/issues/4534 is being raised from this line: https://github.com/chef/chef/blob/master/lib/chef/util/dsc/local_configuration_manager.rb#L107
But we can't raise this as an error here as this method is returning an array which is expected by some methods.
Exception should be caught here: https://github.com/chef/chef/blob/master/lib/chef/provider/dsc_script.rb#L101, but the previous line is not raising any exception even though the PS command fails because `-erroraction` was `continue`

File: spec/unit/util/dsc/local_configuration_manager_spec.rb [hunk 1]
```diff
@@ -59,10 +59,6 @@
         allow(lcm).to receive(:run_configuration_cmdlet).and_return(lcm_status)
       end
       context "that returns successfully" do
-        before(:each) do
-          allow(lcm).to receive(:run_configuration_cmdlet).and_return(lcm_status)
-        end
-
```
Comment (NimishaS) [inline spec/unit/util/dsc/local_configuration_manager_spec.rb]: Removed this as it's written twice. Please expand the code to check.

**Comment by mwrock** [comment]
I'm 👍 on this but please fix style (might just be a matter of rebasing)

**Comment by NimishaS** [comment]
Rebased and signed off.

**Pull request merged**